diff --git a/ultramodern/src/input.cpp b/ultramodern/src/input.cpp
index 363e66c..02cdbd6 100644
--- a/ultramodern/src/input.cpp
+++ b/ultramodern/src/input.cpp
@@ -178,6 +178,9 @@ extern "C" void osContGetReadData(OSContPad *data) {
             data[controller].button = buttons;
             data[controller].err_no = 0;
         } else {
+            data[controller].button = 0;
+            data[controller].stick_x = 0;
+            data[controller].stick_y = 0;
             data[controller].err_no =  CONT_NO_RESPONSE_ERROR; // CHNL_ERR_NORESP >> 4
         }
     }
diff --git a/librecomp/src/cont.cpp b/librecomp/src/cont.cpp
index 9f5858e..d4cf18a 100644
--- a/librecomp/src/cont.cpp
+++ b/librecomp/src/cont.cpp
@@ -50,11 +50,9 @@ extern "C" void osContGetReadData_recomp(uint8_t* rdram, recomp_context* ctx) {
     osContGetReadData(dummy_data);
 
     for (int controller = 0; controller < MAXCONTROLLERS; controller++) {
-        if (dummy_data[controller].err_no == 0) {
-            MEM_H(6 * controller + 0, data) = dummy_data[controller].button;
-            MEM_B(6 * controller + 2, data) = dummy_data[controller].stick_x;
-            MEM_B(6 * controller + 3, data) = dummy_data[controller].stick_y;
-            MEM_B(6 * controller + 4, data) = dummy_data[controller].err_no;
-        }
+        MEM_H(6 * controller + 0, data) = dummy_data[controller].button;
+        MEM_B(6 * controller + 2, data) = dummy_data[controller].stick_x;
+        MEM_B(6 * controller + 3, data) = dummy_data[controller].stick_y;
+        MEM_B(6 * controller + 4, data) = dummy_data[controller].err_no;
     }
 }
